actioncable/test/stubs/test_connection.rb

```diff
@@ -3,24 +3,31 @@
 class TestConnection
   attr_reader :identifiers, :logger, :current_user, :server, :transmissions
 
-  def initialize(user = User.new("lifo"))
+  delegate :pubsub, to: :server
+
+  def initialize(user = User.new("lifo"), coder: ActiveSupport::JSON, subscription_adapter: SuccessAdapter)
+    @coder = coder
     @identifiers = [ :current_user ]
 
     @current_user = user
     @logger = ActiveSupport::TaggedLogging.new ActiveSupport::Logger.new(StringIO.new)
-    @server = TestServer.new
+    @server = TestServer.new(subscription_adapter: subscription_adapter)
     @transmissions = []
   end
 
-  def pubsub
-    SuccessAdapter.new(server)
+  def transmit(cable_message)
+    @transmissions << encode(cable_message)
   end
 
-  def transmit(data)
-    @transmissions << data
+  def last_transmission
+    decode @transmissions.last
   end
 
-  def last_transmission
-    @transmissions.last
+  def decode(websocket_message)
+    @coder.decode websocket_message if websocket_message
```

Comment:
Confirm. Due to trying to decode the last transmission when there were none, and hence decoding `nil`. Fixed.